util/conditions/getter_test.go

```diff
@@ -120,7 +120,7 @@ func TestMirror(t *testing.T) {
 		t.Run(tt.name, func(t *testing.T) {
 			g := NewWithT(t)
 
-			got := Mirror(tt.from, tt.t)
+			got := mirror(tt.from, tt.t)
```

Comment:
We should move these tests to be on the setters now that these methods are private

Reply:
I think that the current code organization also makes sense because the part of code that gets a Mirror condition is in `getter.go`/`getter_test.go` (no matter if they are private), and the set helpers using the mirror condition in `setter.go`

Reply:
Yeah but the general reason you don't test private methods is that they are just an implementation detail of the public methods. They can essentially be inlined into the setters, since they effectively don't exist except for callers within the same package.

Reply:
I think we're also missing coverage of the setters

Reply:
@benmoss do you think that is something that we can handle as a followup or do you consider it a blocker for this PR?

Reply:
I've asked Fabrizio to make these methods private because they might be misleading to users, or hard to use; I don't necessarily mind keeping the tests for now, we can clean it up later

Reply:
>  the general reason you don't test private methods is that they are just an implementation detail of the public methods.

This is a general concern, not strictly related to this PR, but however IMO tests are always a good thing, no matter if for private or public func, and AFAIK we already have a lot of tests on private methods. Also testing private methods helps for separation of concern in unit tests.

> I think we're also missing coverage of the setters

Done

Reply:
Yeah it's not a blocker, just a conversation I thought was worth having